core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/impl/api/schema.kt

```diff
@@ -0,0 +1,54 @@
+package org.jetbrains.kotlinx.dataframe.impl.api
+
+import org.jetbrains.kotlinx.dataframe.columns.ColumnPath
+import org.jetbrains.kotlinx.dataframe.impl.schema.DataFrameSchemaImpl
+import org.jetbrains.kotlinx.dataframe.impl.schema.getSchema
+import org.jetbrains.kotlinx.dataframe.schema.ColumnSchema
+import org.jetbrains.kotlinx.dataframe.schema.DataFrameSchema
+import kotlin.reflect.KClass
+
+@PublishedApi
+internal fun compileTimeSchemaImpl(runtimeSchema: DataFrameSchema, klass: KClass<*>): DataFrameSchema {
+    val compileSchema = getSchema(klass)
+    val root = ColumnPath(emptyList())
+    val order = mutableMapOf<ColumnPath, Int>()
+    runtimeSchema.putColumnsOrder(order, path = root)
+    return compileSchema.sorted(order, path = root)
+}
+
+internal fun DataFrameSchema.putColumnsOrder(order: MutableMap<ColumnPath, Int>, path: ColumnPath) {
```

Comment:
kinda strange to see `DataFrameSchema` be the receiver. I'd expect the mutable map to be the receiver, since that's what you're modifying. That could also enable the notation:
```kt
val order = buildMap<ColumnPath, Int> {
    putColumnsInOrderOf(runtimeSchema, path)
}
```

Reply:
true, i changed it